app/src/main/java/com/example/photogalery/PhotoPageFragment.kt

```diff
@@ -0,0 +1,76 @@
+package com.example.photogalery
+
+import android.annotation.SuppressLint
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebChromeClient
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import androidx.appcompat.app.AppCompatActivity
+import com.example.photogalery.databinding.FragmentPhotoPageBinding
+
+class PhotoPageFragment : PollNotificationHandlerFragment(R.layout.fragment_photo_page) {
+
+    private lateinit var uri: Uri
+    private lateinit var binding: FragmentPhotoPageBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        uri = arguments?.getParcelable(ARG_URI) ?: Uri.EMPTY
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentPhotoPageBinding.inflate(inflater, container, false)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        loadingIntoWebView()
+    }
+
+    @SuppressLint("SetJavaScriptEnabled")
+    private fun loadingIntoWebView() {
+        binding.progressBar.max = 100
+
+        binding.webView.settings.javaScriptEnabled = true
+        binding.webView.webChromeClient = object : WebChromeClient() {
```

Comment:
Whenever making multiple changes on same object we can use `apply` to avoid copy pasting the object over and over again. In this case, `webView.apply {settings.javaScriptEnabled = true .. etc}`